FinalProject/URLValidatorInCorrect/src/UrlValidatorTest.java

```diff
@@ -36,45 +36,360 @@ public UrlValidatorTest(String testName) {
+	   // scheme, host, port, path
+	   System.out.println("Expect true for URL with scheme, host, port, and path - " +
+			   urlVal.isValid("https://www.google.co.uk:80/path/to/something.php"));
+
+	   // scheme, host, port, path, fragment
+	   System.out.println("Expect true for URL with scheme, host, port, path, and fragment - " +
+			   urlVal.isValid("http://www.amazon.com:100/here/is/a/page.aspx#/this/is/a/fragment"));
+
+	   // scheme, host, port, path, query
+	   System.out.println("Expect true for URL with scheme, host, port, path, and query - " +
+			   urlVal.isValid("https://cars.example.com:55/search/for/stuff?type=awesome&wheels=3"));
+
+	   // scheme, host, port, path, query, fragment
+	   System.out.println("Expect true for URL with scheme, host, port, path, query, and fragment - " +
+			   urlVal.isValid("http://bread.plate.ca:6459/here/is/some/cheese/bread?would=you&like=some#yes/please"));
+
+	   // scheme, host, port, query
+	   System.out.println("Expect true for URL with scheme, host, port, and query - " +
+			   urlVal.isValid("about://tell.me.com:169?what=should&this=be"));
+
+	   // scheme, host, port, query, fragment
+	   System.out.println("Expect true for url with scheme, host, port, query, and fragment - " +
+			   urlVal.isValid("http://www.dogbone.com:555/?give=me&a=bone#PLEASE"));
+
+	   // scheme, host, path
+	   System.out.println("Expect true for URL with scheme, host, and path - " +
+			   urlVal.isValid("ftp://sub.domain.com/path/to/file.php"));
+
+	   // scheme, host, path, fragment
+	   System.out.println("Expect true for URL with scheme, host, path, and fragment - " +
+			   urlVal.isValid("http://eat.chicken.com/blargh/pop.aspx#coolThing"));
+
+	   // scheme, host, query
+	   System.out.println("Expect true for URL with scheme, host, and query - " +
+			   urlVal.isValid("https://www.example.com?can=I&has=a&passing=test"));
+
+	   // scheme, host, query, fragment
+	   System.out.println("Expect true for URL with scheme, host, query, and fragment - " +
+			   urlVal.isValid("https://eat.chicken.com?is=this&a=query#woah/its/a/query"));
+
+	   // scheme, host, path, query
+	   System.out.println("Expect true for URL with scheme, host, path, and query - " +
+			   urlVal.isValid("http://www.google.co.uk/take/me/to/your/leader.aspx?foo=bar"));
+
+	   // scheme, host, path, query, fragment
+	   System.out.println("Expect true for URL with scheme, host, path, query, and fragment - " +
+			   urlVal.isValid("https://bear.hug.co.uk/lets/go/dance.php?can=i&has=pears#no/not/ever"));
    }
-   
-   public void testYourSecondPartition(){
-	   
+
+   public void testBadURLsPartition()
+   {
+	   // create our URL validator
+	   UrlValidator urlVal = new UrlValidator(null, null, UrlValidator.ALLOW_ALL_SCHEMES);
+
+	   System.out.println("\nTesting known BAD URLs\n");
+
+	   // scheme, host
+
+	   System.out.println("Expect false for URL with  *scheme and host - "
+			   + urlVal.isValid("totallyBogus://www.example.com"));
```

Comment:
I think that if ALLOW_ALL_SCHEMES is turned on, the regex in UrlValidator should pass a scheme like this. Carlos and I were trying to figure out whether we should be only testing web-related schemes or all schemes.